relay-server/src/extractors/forwarded_for.rs

```diff
@@ -11,12 +11,14 @@ pub struct ForwardedFor(String);
 impl ForwardedFor {
     const FORWARDED_HEADER: &str = "X-Forwarded-For";
     const VERCEL_FORWARDED_HEADER: &str = "X-Vercel-Forwarded-For";
+    const SENTRY_FORWARDED_HEADER: &str = "X-Sentry-Forwarded-For";
 
     /// We prefer the Vercel header because the normal one could get overwritten as explained here.
     /// `https://vercel.com/docs/concepts/edge-network/headers#x-vercel-forwarded-for`
     fn get_forwarded_for_ip(header_map: &HeaderMap) -> &str {
         header_map
             .get(Self::VERCEL_FORWARDED_HEADER)
+            .or_else(|| header_map.get(Self::SENTRY_FORWARDED_HEADER))
```

Comment:
The question came up if we should value our own header over the vercel header, the conclusion was:

> 1. our infrastructure requires the "standard" x-forwarded-for to be trusted, so it puts the contents of just that header into the sentry version.
> 2. at the same time, x-vercel-forwarded-for is the more accurate header overall if it is there. so it should be preferred over the other two.

Can you add the explanation as a comment so somebody in the future knows why the sentry header is not the highest priority (like there is already an explanation why the vercel header is preferred over the standard X-Forwarded-For header)?

Reply:
Done with 3a97475d1e6d11ad6724c935a39e62fa61489582